system_setup/server/controllers/configure.py

```diff
@@ -37,6 +41,208 @@ def __init__(self, app):
+                      clsCommand)
+            return True
+
+        cache = apt.Cache()
+        if self.app.opts.dry_run:
+            packs_dir = os.path.join(self.model.root,
+                                     apt_pkg.config
+                                     .find_dir("Dir::Cache::Archives")[1:])
+            os.makedirs(packs_dir, exist_ok=True)
+            try:
+                for package in packages:
+                    # Just write the package uri to a file.
+                    archive = os.path.join(packs_dir, cache[package].fullname)
+                    with open(archive, "wt") as f:
+                        f.write(cache[package].candidate.uri)
+            except IOError:
+                log.error("Failed to write %s file.", archive)
+                return False
+
+            return True
+
+        cache.update()
+        cache.open(None)
+        with cache.actiongroup():
+            for package in packages:
+                cache[package].mark_install()
+
+            cache.commit()
+
+        return True
+
+    def _update_locale_gen_file(self, localeGenPath, lang) -> bool:
+        """ Uncomments the line in locale.gen file where lang is found,
+            if found commented. A fully qualified language is expected,
+            since that would have passed thru the Locale controller
+            validation. e.g. en_UK.UTF-8. Returns True for success. """
+
+        fileContents: str
+        try:
+            with open(localeGenPath, "r+") as f:
+                fileContents = f.read()
+                lineFound = fileContents.find(lang)
+                if lineFound == -1:
+                    # An unsupported locale coming from our UI is a bug
+                    log.error("Selected language %s not supported.", lang)
+                    return False
+
+                pattern = r'[# ]*({}.*)'.format(lang)
+                commented = re.compile(pattern)
+                (fileContents, n) = commented.subn(r'\1', fileContents,
+                                                   count=1)
+                if n != 1:
+                    log.error("Unexpected locale.gen file contents. Aborting.")
+                    return False
+
+                f.seek(0)
+                f.write(fileContents)
+                return True
+
+        except IOError as err:
+            log.error("Failed to modify %s file. %s", localeGenPath, err)
+            return False
+
+    def _locale_gen_file_path(self):
+        localeGenPath = "/etc/locale.gen"
+        if self.app.opts.dry_run is False or self.app.opts.dry_run is None:
+            return localeGenPath
+
+        # For testing purposes.
+        etc_dir = os.path.join(self.model.root, "etc")
+        testLocGenPath = os.path.join(etc_dir,
+                                      os.path.basename(localeGenPath))
+        shutil.copy(localeGenPath, testLocGenPath)
+        shutil.copy(localeGenPath, "{}-".format(testLocGenPath))
```

Comment:
We generally use `.new` for this ? This is what we are generally using to have atomic changes. (ok, here, it’s contrived because it’s only for tests, but maybe .test?)